inference-engine/src/mkldnn_plugin/nodes/common/jit_load_store_emitters.cpp

```diff
@@ -0,0 +1,887 @@
+//
+
+#include "emitter.h"
+#include "jit_load_store_emitters.h"
+#include "legacy/ie_layers.h"
+#include "jit_generator.hpp"
+#include "utils/bfloat16.hpp"
+
+using namespace InferenceEngine;
+using namespace mkldnn::impl;
+using namespace mkldnn::impl::utils;
+using namespace mkldnn::impl::cpu;
+using namespace Xbyak;
+using namespace Xbyak::util;
+
+namespace MKLDNNPlugin {
+
+/// LOAD ///
+jit_load_emitter::jit_load_emitter(jit_generator *host, cpu_isa_t host_isa, const MKLDNNNode* node, Precision exec_prc)
+: jit_emitter(host, host_isa, node, exec_prc) {
+    prepare_table();
+    v_len_elt = get_vec_length() / exec_prc.size();
+}
+
+void jit_load_emitter::emit_impl(const std::vector<size_t> &in_idxs, const std::vector<size_t> &out_idxs,
+                  const std::vector<size_t> &pool_vec_idxs, const std::vector<size_t> &pool_gpr_idxs,
+                  const emitter_context *emit_context) {
+    const auto* load_emitter_context = dynamic_cast<const MKLDNNPlugin::load_emitter_context*>(emit_context);
+    if (load_emitter_context == nullptr) {
+        THROW_IE_EXCEPTION << "Load " << n->getName() << " does not get load emmiter context.";
+    }
+    if (host_isa_ == cpu::sse42) {
+        emit_isa<cpu::sse42>(Reg64(in_idxs[0]), load_emitter_context->offset_byte_, load_emitter_context->src_prc_, out_idxs[0],
+            load_emitter_context->load_num_, load_emitter_context->is_fill_, load_emitter_context->fill_value_);
+    } else if (host_isa_ == cpu::avx2) {
+        emit_isa<cpu::avx2>(Reg64(in_idxs[0]), load_emitter_context->offset_byte_, load_emitter_context->src_prc_, out_idxs[0],
+            load_emitter_context->load_num_, load_emitter_context->is_fill_, load_emitter_context->fill_value_);
+    } else if (host_isa_ == cpu::avx512_common) {
+        emit_isa<cpu::avx512_common>(Reg64(in_idxs[0]), load_emitter_context->offset_byte_, load_emitter_context->src_prc_, out_idxs[0],
+            load_emitter_context->load_num_, load_emitter_context->is_fill_, load_emitter_context->fill_value_);
+    } else {
+        assert(!"unsupported isa");
+    }
+}
+
+template <mkldnn::impl::cpu::cpu_isa_t isa>
+void jit_load_emitter::emit_isa(const Xbyak::Reg64 &reg_src, size_t offset_byte, InferenceEngine::Precision src_prc,
+    const int out_vec_idx, int load_num, bool is_fill, std::string fill_value) const {
+    if ((isa == cpu::sse42 && load_num > 4) || (isa == cpu::avx2 && load_num > 8) ||
+        (isa == cpu::avx512_common && load_num > 16) || load_num < 0) {
+        assert(!"unexpected number of elements to load");
+    }
+    using Vmm = typename conditional3<isa == cpu::sse42, Xmm, isa == cpu::avx2, Ymm, Zmm>::type;
+
+    constexpr bool is_zmm = std::is_same<Vmm, Xbyak::Zmm>::value;
+    constexpr bool is_ymm = std::is_same<Vmm, Xbyak::Ymm>::value;
+    constexpr bool is_xmm = std::is_same<Vmm, Xbyak::Xmm>::value;
+
+    if (is_xmm) {
+        switch (src_prc) {
+            case Precision::FP32:
```

Comment:
It looks like here and in the next if statement you use completely the same set of calls.  May be it would be better to call load_bytes, load_bytes_to_dword_extension etc. for YMM and XMM polymorphically, passing VMM as a template parameter for example.

Reply:
combined for xmm/ymm/zmm. Thank you.